### DISCUSSION TITLE
ROUTING_INFEASIBLE vs ROUTING_FAIL in CVRPTW

### DISCUSSION THREAD

**Discussion by onecable5781** [google/or-tools#4182] [Routing (and legacy CP) questions]
Hello,

I have the following code, https://godbolt.org/z/qKx8TbvPW which reads in the following input file:

[input.txt](https://github.com/google/or-tools/files/14970709/input.txt)

The problem is a depot + 2 customer (3 nodes in total, first column of first row in input file) CVRPTW problem. There is a single vehicle with 10 units of capacity (2nd and 3rd columns of first row of input file) being solved for 200,000 milli seconds (next entry in first row).
The demand at customer 1 is 20 units and demand at customer 2 is 30 units (please see last columns of ultimate and penultimate rows of input file). So, the problem is clearly infeasible.

When this problem is solved, I check the status of the solution via the following lines:

```
if (routing.status() == routing.ROUTING_INFEASIBLE)
		actuallyinfeasible = 1;
if (routing.status() == routing.ROUTING_FAIL || routing.status() == routing.ROUTING_FAIL_TIMEOUT)
		infeasible = 1;
```

The returned status is `routing.ROUTING_FAIL`. Relevant status are:

https://github.com/google/or-tools/blob/5f7eabb3ffa1de786c224a722f0e9561101c3ab2/ortools/constraint_solver/routing.h#L268-L275

Can it please be clarified why this problem is not being deemed to have proven infeasibility? That is more useful in this context than the heuristic being unable to find a feasible solution. Is there a way the solver can be encouraged to prove infeasibility vs finding a good heuristic solution? If infeasibility is proven, then, the latter can essentially be skipped.

Thank you.

**Comment by lperron** [comment]
From what I believe, the solver is not built to prove infeasibility. It will only return INFEASIBLE if it is trivially infeasible, not after a search.
